### PR TITLE
DOC: Improved the docstring of pandas.Series.filter

### PR DESCRIPTION
**Pull request by ottiP**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
####################### Docstring (pandas.Series.filter) #######################
################################################################################

Subset rows or columns of dataframe according to labels in
the specified index.

Note that this routine does not filter a dataframe on its
contents. The filter is applied to the labels of the index.

Parameters
----------
items : list-like
    List of info axis to restrict to (must not all be present).
like : string
    Keep info axis where "arg in col == True".
regex : string (regular expression)
    Keep info axis with re.search(regex, col) == True.
axis : int or string axis name
    The axis to filter on.  By default this is the info axis,
    'index' for Series, 'columns' for DataFrame.

Returns
-------
same type as input object

Examples
--------
>>> df = pd.DataFrame(np.array(([1,2,3],[4,5,6])),
...                   index=['mouse', 'rabbit'],
...                   columns=['one', 'two', 'three'])

>>> # select columns by name
>>> df.filter(items=['one', 'three'])
         one  three
mouse     1      3
rabbit    4      6

>>> # select columns by regular expression
>>> df.filter(regex='e$', axis=1)
         one  three
mouse     1      3
rabbit    4      6

>>> # select rows containing 'bbi'
>>> df.filter(like='bbi', axis=0)
         one  two  three
rabbit    4    5      6

See Also
--------
pandas.DataFrame.loc : Purely label-location based indexer for selection by label.

Notes
-----
The ``items``, ``like``, and ``regex`` parameters are
enforced to be mutually exclusive.

``axis`` defaults to the info axis that is used when indexing
with ``[]``.

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Series.filter" correct. :)

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3525,46 +3525,45 @@ def filter(self, items=None, like=None, regex=None, axis=None):
         Parameters
         ----------
         items : list-like
-            List of info axis to restrict to (must not all be present)
+            List of info axis to restrict to (must not all be present).
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Hmm I don't think `info axis` is well known. How about "List of labels from `axis` to restrict to."

File: pandas/core/generic.py [hunk 2]
```diff
@@ -3525,46 +3525,45 @@ def filter(self, items=None, like=None, regex=None, axis=None):
...
         -------
         same type as input object
 
         Examples
         --------
-        >>> df
-        one  two  three
-        mouse     1    2      3
-        rabbit    4    5      6
+        >>> df = pd.DataFrame(np.array(([1,2,3],[4,5,6])),
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: PEP8: spaces after comma

File: pandas/core/generic.py (hunk 1 above)
Comment (TomAugspurger) [reply pandas/core/generic.py]: And maybe replace all other "info axis" with \`axis\`

File: pandas/core/generic.py [hunk 3]
```diff
@@ -3525,46 +3525,45 @@ def filter(self, items=None, like=None, regex=None, axis=None):
...
         >>> # select rows containing 'bbi'
         >>> df.filter(like='bbi', axis=0)
-        one  two  three
+                 one  two  three
         rabbit    4    5      6
 
         See Also
         --------
-        pandas.DataFrame.loc
+        pandas.DataFrame.loc : Purely label-location based indexer for selection by label.
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Maybe too long. Just "label-based indexer."

File: pandas/core/generic.py [hunk 4]
```diff
@@ -3794,6 +3793,19 @@ def sample(self, n=None, frac=None, replace=False, weights=None,
         40  0.823173 -0.078816  1.009536  1.015108
         15  1.421154 -0.055301 -1.922594 -0.019696
         6  -0.148339  0.832938  1.787600 -1.383767
+        
+        You can use `random state` for reproducibility:
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Is this from a different PR?

**Comment by jreback** [comment]
@datapythonista if you want to merge this on green.

**Comment by jreback** [comment]
thanks @ottiP !

**Pull request merged**